### PR TITLE
Remove explicit announce from local replica in ReadFromMergeTree

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 42670a46d49719efc5caae1ca23b6d95360fd02d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69602/42670a46d49719efc5caae1ca23b6d95360fd02d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/2fce90ab76b851198b39041c9e7c4b03ad601ff7/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/integration/test_parallel_replicas_snapshot_from_initiator/test.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        pytest.skip("Disabled for sanitizers (too slow)")
+
+    table_name = "t"
+    _create_tables(table_name)
+
+    for idx, node in enumerate(nodes):
+        expected = 499500 + idx  # sum of all integers 0..999 + idx
+        assert (
+            node.query(
+                "SELECT sum(value) FROM t",
```
Comment (devcrafter) [inline tests/integration/test_parallel_replicas_snapshot_from_initiator/test.py:65]:
```suggestion
                "SELECT sum(value) FROM {table_name}",
```

File: tests/integration/test_parallel_replicas_snapshot_from_initiator/test.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    for idx, node in enumerate(nodes):
+        node.query("SYSTEM STOP REPLICATED SENDS")
+        # the same data on all nodes except for a single value
+        node.query(
+            f"INSERT INTO {table_name} VALUES ({idx})",
+            settings={"insert_deduplicate": 0},
+        )
+
+
+def test_number_of_marks_read(start_cluster):
```
Comment (devcrafter) [inline tests/integration/test_parallel_replicas_snapshot_from_initiator/test.py:54]: It'd always be nice to have a comment about what test supposed to check

**Review by devcrafter — APPROVED** [review]

**Comment by nickitat** [comment]
`Stress test (debug)` - #69573

**Pull request merged**